### PR TITLE
changes to partially sync svn

### PR DESCRIPTION
**Pull request by mlzummo**
@jbrundage @GordonSmith Please Review

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/api/INDChart.js [hunk 1]
```diff
@@ -12,13 +12,28 @@
...
     //  Data ---
     INDChart.prototype.testData = function () {
-        this.columns(["Subject", "Year 1", "Year 2", "Year 3"]);
-        this.data([
-            ["Geography", 75, 68, 65],
-            ["English", 45, 55, 52],
-            ["Math", 98, 92, 90],
-            ["Science", 66, 60, 66]
-        ]);
+        switch (this._chartType) {
```
Comment (GordonSmith) [inline src/api/INDChart.js]: Does this work for charts with "_chartType" of undefined?

File: src/chart/Pie.js [hunk 2]
```diff
@@ -31,7 +31,7 @@
     Pie.prototype._class += " chart_Pie";
     Pie.prototype.implements(I2DChart.prototype);
 
-    Pie.prototype.publish("paletteID", "default", "set", "Palette ID", Pie.prototype._palette.switch());
+    Pie.prototype.publish("paletteID", "default", "set", "Palette ID", Pie.prototype._palette.switch(),{tags:['Basic','Shared']});
```
Comment (GordonSmith) [inline src/chart/Pie.js]: @jbrundage What does the "Shared" mean?

File: src/marshaller/HipieDDL.js [hunk 3]
```diff
@@ -52,17 +52,17 @@
 
     SourceMappings.prototype.doMap = function (item) {
         var retVal = [];
-        for (var key in this.mappings) {
-            try {
+        try {
+            for (var key in this.mappings) {
```
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: You are rolling back newer changes...

**Comment by GordonSmith** [comment]
At least one change was reverting a newer commit - can you double check the others?

File: src/api/INDChart.js (hunk 1 above)
Comment (mlzummo) [inline src/api/INDChart.js]: yes it does, just tested

File: src/marshaller/HipieDDL.js (hunk 3 above)
Comment (mlzummo) [inline src/marshaller/HipieDDL.js]: i wasn't sure about that one, i didn't know if we wantted the try in the loop or outside, reverting will make the try catch in the loop

Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: Yes - I switched that way so I could provide better error reporting

File: src/chart/Pie.js (hunk 2 above)
Comment (mlzummo) [inline src/chart/Pie.js]: I believe we are going with its shared among 1 or more widgets (not ness. across libs)

**Comment by mlzummo** [comment]
@GordonSmith  i "cherry picked" all the changes from SVN -> GIT so, we should be good to go .. that one change that reverted was me cherry picking that change.

File: src/chart/Pie.js (hunk 2 above)
Comment (GordonSmith) [inline src/chart/Pie.js]: Can this not be calculated on the fly?  Otherwise it will for ever be accidentally wrong (what is it actually used for?)

**Comment by GordonSmith** [comment]
@mlzummo Are these "fixes" for 1.0.0 or are they new features for 1.1.0?

**Comment by mlzummo** [comment]
@GordonSmith Fixes for 1.0.0, i rebased this against master though, i want to make sure this doesn't break anything on 1.0.0 ... which branch can i rebase against ?

**Comment by GordonSmith** [comment]
Use candidate-1.0.2 for bug fixes to 1.0.0 and use master for the "next" build

**Pull request merged**